File: src/my_project/easy_problems/from1to50/Implement strStr.py

if __name__ == '__main__':

    class Solution:
        def strStr(self, haystack: str, needle: str) -> int:
            index = -1
            len_needle = len(needle)
            flag = None

            if len(haystack) == 0 and len_needle == 0:
                return 0
            elif len(haystack) == 0 and len_needle == 0:
                return 0
            elif len(haystack) == 0 and len_needle != 0:
                return -1
            else:
                for i in range(len(haystack)):
                    flag = True
                    if haystack[i] == needle[0] and len(haystack[i:]) >= len_needle:
                        index = i
                        for j in range(len_needle):
                            if haystack[i+j] != needle[j]:
                                index = -1
                                flag = False
                                break
                        if flag == True:
                            return index

                return index

    temp_list = [1,2,3,4]
    print(temp_list[3:])

    solution = Solution()
    print(solution.strStr("aaaa","aa"))

File: my_project_typescript/easy_problems_typescript/from1to50/implementstrStr.ts
function strStr(haystack: string, needle: string): number {
  let needleLenght: number = needle.length;
  let flag: boolean;
  let index: number = -1;
  if (needleLenght === 0) {
    return 0;
  } else if (haystack.length === 0 && needleLenght !== 0) {
    return -1;
  } else {
    for (let i = 0; i < haystack.length; i++) {
      flag = true;
      if (
        haystack[i] === needle[0] &&
        haystack.slice(i).length >= needleLenght
      ) {

        index = i;
        for (let j = 0; j < needleLenght; j++) {
   
          if (needle[j] !== haystack[i + j]) {
            index = -1;
            flag = false;
            break;
          }

        }
        if (flag) {
            return index;
          }
      }
    }
    return index;
  }
}

// console.log(strStr("hello", "ll"));
// console.log(strStr("aaaaa", "bba"));
strStr("mississippi","issip");
